lib/Graph/NodeValue.cpp

```diff
@@ -36,7 +36,17 @@ NodeValue::NodeValue(Node *N, unsigned resNo) {
 void NodeValue::replaceAllUsesOfWith(NodeValue v, const Function *F,
                                      Node *skipReplacement) const {
   if (v.getNode()) {
-    assert(getType() == v.getType() && "Replacing value with the wrong type");
+    if (getType() != v.getType()) {
```

Comment:
nit: could fold this condition into above if, and remove one indentation level here.